components/board/genericlinux/gpio.go

```diff
@@ -232,5 +253,72 @@ func gpioInitialize(cancelCtx context.Context, gpioMappings map[int]GPIOBoardMap
 			logger:     logger,
 		}
 	}
-	return pins
+	return pins, interrupts, nil
+}
+
+type digitalInterrupt struct {
+    interrupt  board.DigitalInterrupt
+	line       *gpio.LineWithEvent
+    cancelCtx  context.Context
+	cancelFunc func()
+}
+
+func createDigitalInterrupt(ctx context.Context, config board.DigitalInterruptConfig, gpioMappings map[int]GPIOBoardMapping, activeBackgroundWorkers *sync.WaitGroup) (*digitalInterrupt, error) {
+	pinInt, err := strconv.Atoi(config.Pin)
+	if err != nil {
+		return nil, errors.Errorf("pin names must be numerical")
+	}
+	mapping, ok := gpioMappings[pinInt]
+	if !ok {
+		return nil, errors.Errorf("Unknown interrupt pin %s", config.Pin)
+	}
+
+	chip, err := gpio.OpenChip(mapping.GPIOChipDev)
+	if err != nil {
+		return nil, err
+	}
+	defer utils.UncheckedErrorFunc(chip.Close)
+
+	// TODO: do we need to configure the line to be open-drain or open-source?
```

Comment:
I don't know the answer, but I'd like other opinions. Is there something we should do with this?

Reply:
It looks like push-pull is the default GPIO setting, according to this [source](https://embeddedartistry.com/blog/2018/06/04/demystifying-microcontroller-gpio-settings/) I've been reading from. So my two cents is to leave the default for now, and note somewhere that we don't currently expose the ability to configure otherwise?

Reply:
Thanks for the link! It looks like this is much more important for outputs than inputs, and the comment belongs in the old code for basic GPIO functionality. I'll put it there instead.